Sources/Plasma/FeatureLib/pfPython/cyMiscGlue4.cpp

```diff
@@ -405,34 +405,50 @@ PYTHON_GLOBAL_METHOD_DEFINITION(PtDebugAssert, args, "Params: cond, msg\nDebug o
 PYTHON_GLOBAL_METHOD_DEFINITION_WKEY(PtDebugPrint, args, kwargs, "Params: *msgs, **kwargs\nPrints msgs to the Python log given the message's level")
 {
     uint32_t  level = cyMisc::kErrorLevel;
+    ST::string sep = ST_LITERAL(" ");
 
     do {
         // Grabbin' levelz
         if (kwargs && PyDict_Check(kwargs)) {
-            PyObject* value = PyDict_GetItem(kwargs, PyString_FromString("level"));
+            PyObject* value = PyDict_GetItemString(kwargs, "level");
             if (value) {
                 if (PyInt_Check(value))
                     level = PyInt_AsLong(value);
                 else
                     break;
             }
+
+            value = PyDict_GetItemString(kwargs, "sep");
+            if (value) {
+                if (PyString_CheckEx(value))
+                    sep = PyString_AsStringEx(value);
+                else
+                    break;
+            }
         }
 
+        ST::string_stream ss;
         for (size_t i = 0; i < PySequence_Fast_GET_SIZE(args); ++i) {
             PyObject* theMsg = PySequence_Fast_GET_ITEM(args, i);
-            if (!PyString_CheckEx(theMsg))
-                theMsg = PyObject_Repr(theMsg);
+            if (PyString_CheckEx(theMsg))
+                Py_XINCREF(theMsg);
+            else
+                theMsg = PyObject_Str(theMsg);
 
+            if (i != 0)
+                ss << sep;
             if (theMsg)
-                cyMisc::DebugPrint(PyString_AsStringEx(theMsg), level);
+                ss << PyString_AsStringEx(theMsg);
             else
-                break;
+                ss << "<ERROR>";
```

Comment:
I feel like the previous use of `break` was intended to cause us to get to the `PyError_SetString` code outside of the `do ... while (false)` block (however, due to the inner loop, it didn't actually do that)...  Perhaps we should fix that rather than dumping `"<ERROR>"` to the log?  Especially since a null `theMsg` should indicate that an error was already set by another Python API...